### DISCUSSION TITLE
New column based on NumPy using current column as a list/array?

### DISCUSSION THREAD

**Discussion by wbmphd** [saulpw/visidata#2156] [Q&A] [answered]
I would like to create a new column that is the result of `numpy.diff(<current column as a 1-D array>)`.  Is this possible?

In other words, is it possible to treat the current column as a list in the creation of a new column.  From my (limited) research, it appears that all operations are cell-by-cell, which would prohibit this.  Can anyone educate me?

Thanks in advance!

**Comment by saulpw** [comment, marked as answer]
Hi @wbmphd, I don't think you can quite use `numpy.diff`, but you can get a windowed view of the current column; use the `w` command and specify the number of rows before/after you want in each list.  Then you can use an expression column to do what you want with that last.

**Reply by wbmphd** [reply]
Thank you for the reply, Saul!  And also thank you for such a wonderful tool.
